### PR TITLE
[WL5] Import State Data

### LINKED ISSUE #6: [WL5] Import State Table
Import the State Table into a new Schema+Model + any hard dependency tables

Child of #5

### PR DESCRIPTION
**Pull request by PeterDKC**
- Refactor Import Command
- Refactor State Model & some relational things
- Remove setter cruft
- Imports State data into State Model

Resolves #6 

image

### CODE REVIEW

File: app/Console/Commands/LegiScanImport.php [hunk 1]
```diff
@@ -2,58 +2,96 @@
...
-        $command .= ' PORT='.config('database.connections.mysql.port');
-        $command .= ' NAME='.config('database.connections.mysql.database');
-        $command .= ' USER='.config('database.connections.mysql.username');
-        $command .= ' PASS='.config('database.connections.mysql.password');
+        foreach ($states as $state) {
+            State::updateOrCreate(
+                ['id' => $state->state_id],
+                [
+                    'id'           => $state->state_id,
+                    'name'         => $state->state_name,
```
[the remark this answers is not in the document]
Comment (PeterDKC) [reply app/Console/Commands/LegiScanImport.php:50]: Isn't this the homepage of the current site?...


image

File: tests/Unit/Models/LegiScan/StateTest.php [hunk 2]
```diff
@@ -13,7 +13,7 @@ public function statesHaveManyBills()
     {
         $state = State::factory()->create();
         Bill::factory()->create([
-            'state' => $state,
+            'state_id'    => $state->id,
```
Comment (barobba) [inline tests/Unit/Models/LegiScan/StateTest.php:16]: I assume we won't need to use `state_id` once the bill is updated to our own models, like the state was?

**Review by barobba — COMMENTED** [review]
Looks good!

File: app/Console/Commands/LegiScanImport.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (barobba) [reply app/Console/Commands/LegiScanImport.php:50]: I wouldn't go by the homepage. I would imagine instead, a data scientist querying the entire data set for trends to share. For example, we will eventually have things like vote counts... and make much more use of the other table data.

If we want to import anything we would conceivably want, then we should probably just import all the columns, including capitol name (which I can easily see us making use of) and the coordinates. I am curious if the coordinates are for the capitol building, in which case we would definitely want that.

[the remark this answers is not in the document]
Comment (barobba) [reply app/Console/Commands/LegiScanImport.php:50]: You also excluded `biennium` and `carry_over`. Why?

[the remark this answers is not in the document]
Comment (PeterDKC) [reply app/Console/Commands/LegiScanImport.php:50]: Data scientists at a minimum mediocre at SQL, I wouldn't worry about it.

Adding in the extra fields.

File: app/Traits/Models/HasEnumProperties.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+trait HasEnumProperties
+{
+    /**
+     * Enum property getter
+     *
+     * @param string $field
+     * @return mixed|false
+     */
+    public static function getEnum(string $field)
```
[the remark this answers is not in the document]
Comment (barobba) [reply app/Traits/Models/HasEnumProperties.php:16]: https://laravel.com/docs/9.x/releases#enum-casting
https://www.php.net/manual/en/language.enumerations.backed.php

[the remark this answers is not in the document]
Comment (PeterDKC) [reply app/Traits/Models/HasEnumProperties.php:16]: This is a completely different use case than those and nothing to do w/ PHP Enums. This is a just a helper Trait that Jordan and I came up with years ago that I drag around to every app I work on. Explanation here: https://www.youtube.com/watch?v=9QzQowWgnZA&t=363s&ab_channel=BeardAboutIt

**Pull request merged**